plugin/guess_executable_cmake.vim

```diff
@@ -0,0 +1,49 @@
+let cmake_build_dir="./build/"
+let cmake_api_reply_dir=cmake_build_dir . ".cmake/api/v1/reply/"
+function ArtifactsOfLibrary(buffer_name)
+        let targets=split(glob(g:cmake_api_reply_dir . "target*"))
+        " Decode all target_file JSONS into Dictionaries
+        call map(targets, {idx, val -> json_decode(readfile(val))})
+        " Filter down to targets that use buffer_name as a source
+        let Filter_lambda = {idx, val -> match(map(get(get(val, "link", {}),"commandFragments", []), {idx, source -> source.fragment}), a:buffer_name) >= 0}
+        call filter(targets, Filter_lambda ) 
+        " Get all artifacts that buffer_name generates
+        call flatten(map(targets, {idx, target_file -> map(target_file.artifacts, {idx, artifact -> artifact.path})}))
+        return targets
+endfunction
+
+function ArtifactsOfSource(buffer_name)
```

Comment:
Just in case, would it be possible to take advantage of cmake introspection without any buffer opened? For instance, I launch nvim and start debugging right away.

Reply:
I added an command-completion that finds all executable files. I can see if I can also just add all cmake executables to the command list in front of that.